ethportal-peertest/src/lib.rs

```diff
@@ -2,31 +2,75 @@ pub mod cli;
 pub mod jsonrpc;
 
 pub use cli::PeertestConfig;
-pub use jsonrpc::{get_enode, test_jsonrpc_endpoints_over_http, test_jsonrpc_endpoints_over_ipc};
+pub use jsonrpc::get_enode;
 
 use std::sync::Arc;
 
+use futures::future;
+
 use trin_core::cli::TrinConfig;
 use trin_core::jsonrpc::service::JsonRpcExiter;
 
-pub async fn launch_buddy_node() -> (String, Arc<JsonRpcExiter>) {
+pub struct PeertestNode {
+    pub enr: String,
+    pub exiter: Arc<JsonRpcExiter>,
+}
+
+pub struct AllPeertestNodes {
+    pub bootnode: PeertestNode,
+    pub nodes: Vec<PeertestNode>,
+}
+
+impl AllPeertestNodes {
+    pub fn exit(&self) {
+        self.bootnode.exiter.exit();
+        self.nodes.iter().for_each(|node| node.exiter.exit());
+    }
+}
+
+pub async fn launch_node(mut id: u8, bootnode: Option<&String>) -> anyhow::Result<PeertestNode> {
+    // The bootnode's id is 1, the other node ids are subsequently incremented by one
+    if bootnode.is_some() {
+        id += 1;
+    }
     // Run a client, as a buddy peer for ping tests, etc.
-    let trin_config = TrinConfig::new_from(
-        [
+    let discovery_port = format!("900{id}");
```

Comment:
I don't know that we would ever have 10 or more nodes, but that would create an invalid port. We ought to do something like

```rust
let discovery_port: u16 = 9000 + id;
```